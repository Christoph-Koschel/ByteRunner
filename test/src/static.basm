%entry main
%include "./test/src/natives.hasm"

%define Hello "WASD"
%qword MY_INT 255

dump_mem:
    push 0
    push 20
    int dump_memory
    ret

main:
    call dump_mem
    push MY_INT
    read64
    int print_i64

    push MY_INT
    push 7583758
    write64
    call dump_mem

    push MY_INT
    read64
    int print_i64
    halt