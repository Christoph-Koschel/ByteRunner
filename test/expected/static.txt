57 41 53 44 FF 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
255
57 41 53 44 0E B8 73 00 00 00 00 00 00 00 00 00 00 00 00 00 
7583758
